File: challenges/2026-09-26-steady-relay-subtrees/solution.go
package main

import "fmt"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

// buildTree builds a tree from LeetCode-style compact level-order values.
// A nil entry in values represents a missing child.
func buildTree(values []*int) *TreeNode {
	if len(values) == 0 || values[0] == nil {
		return nil
	}
	root := &TreeNode{Val: *values[0]}
	queue := []*TreeNode{root}
	i := 1
	for len(queue) > 0 && i < len(values) {
		node := queue[0]
		queue = queue[1:]
		if i < len(values) {
			if values[i] != nil {
				node.Left = &TreeNode{Val: *values[i]}
				queue = append(queue, node.Left)
			}
			i++
		}
		if i < len(values) {
			if values[i] != nil {
				node.Right = &TreeNode{Val: *values[i]}
				queue = append(queue, node.Right)
			}
			i++
		}
	}
	return root
}

// countSteadySubtrees returns the number of nodes whose subtree is k-steady.
func countSteadySubtrees(root *TreeNode, k int) int {
	// TODO: implement
	return 0
}

func iptr(v int) *int { return &v }

func main() {
	type testCase struct {
		values   []*int
		k        int
		expected int
	}

	tests := []testCase{
		{[]*int{iptr(1), iptr(2), iptr(3), iptr(4), nil}, 1, 4},
		{[]*int{iptr(1), iptr(2), iptr(3), iptr(4), nil}, 0, 2},
		{[]*int{iptr(10), iptr(5), iptr(15), iptr(3), iptr(8), nil, iptr(20), iptr(1)}, 0, 3},
		{[]*int{}, 3, 0},                                              // extra: empty tree
		{[]*int{iptr(5)}, 0, 1},                                       // extra: single node
		{[]*int{iptr(1), iptr(2), nil, iptr(3), nil}, 2, 3},           // extra: left-leaning chain
	}

	for _, tc := range tests {
		root := buildTree(tc.values)
		got := countSteadySubtrees(root, tc.k)
		status := "FAIL"
		if got == tc.expected {
			status = "PASS"
		}
		fmt.Printf("%s k=%d expected=%d got=%d\n", status, tc.k, tc.expected, got)
	}
}

File: challenges/2026-09-26-steady-relay-subtrees/solution.cs
using System;
using System.Collections.Generic;

public class TreeNode
{
    public int Val;
    public TreeNode Left;
    public TreeNode Right;
    public TreeNode(int val = 0, TreeNode left = null, TreeNode right = null)
    {
        Val = val;
        Left = left;
        Right = right;
    }
}

public class Solution
{
    // BuildTree builds a tree from LeetCode-style compact level-order values.
    // A null entry represents a missing child.
    public static TreeNode BuildTree(int?[] values)
    {
        if (values.Length == 0 || values[0] == null) return null;
        var root = new TreeNode(values[0].Value);
        var queue = new Queue<TreeNode>();
        queue.Enqueue(root);
        int i = 1;
        while (queue.Count > 0 && i < values.Length)
        {
            var node = queue.Dequeue();
            if (i < values.Length)
            {
                var v = values[i++];
                if (v != null)
                {
                    node.Left = new TreeNode(v.Value);
                    queue.Enqueue(node.Left);
                }
            }
            if (i < values.Length)
            {
                var v = values[i++];
                if (v != null)
                {
                    node.Right = new TreeNode(v.Value);
                    queue.Enqueue(node.Right);
                }
            }
        }
        return root;
    }

    // Returns the number of nodes whose subtree is k-steady.
    public static int CountSteadySubtrees(TreeNode root, int k)
    {
        // TODO: implement
        return 0;
    }

    public static void Main(string[] args)
    {
        var tests = new List<(int?[] Values, int K, int Expected)>
        {
            (new int?[] {1, 2, 3, 4, null}, 1, 4),
            (new int?[] {1, 2, 3, 4, null}, 0, 2),
            (new int?[] {10, 5, 15, 3, 8, null, 20, 1}, 0, 3),
            (new int?[] {}, 3, 0),                              // extra: empty tree
            (new int?[] {5}, 0, 1),                             // extra: single node
            (new int?[] {1, 2, null, 3, null}, 2, 3),           // extra: left-leaning chain
        };

        foreach (var (values, k, expected) in tests)
        {
            var root = BuildTree(values);
            var got = CountSteadySubtrees(root, k);
            string status = got == expected ? "PASS" : "FAIL";
            Console.WriteLine($"{status} k={k} expected={expected} got={got}");
        }
    }
}
